File: go/0007-reverse-integer.go
package main

// https://leetcode.com/problems/reverse-integer/description/

import (
	"math"
)

func reverse(x int) int {

	reversedNumber := 0

	for x != 0 {
		reversedNumber = reversedNumber*10 + x%10
		x /= 10
	}

	if reversedNumber >= math.MaxInt32 || reversedNumber <= math.MinInt32 {
		return 0
	}

	return reversedNumber

}

// func main() {
// 	fmt.Println(reverse(-123))
// }

File: rust/src/0007-reverse-integer.rs

// https://leetcode.com/problems/reverse-integer/

pub fn reverse(mut x: i32) -> i32 {

    let mut revered= 0;

    while x != 0 {
        revered = revered * 10 + x%10;
        x /= 10;
    }
    
    if revered > std::i32::MAX || revered < std::i32::MIN {
        return 0
    }

    revered
}

fn main() {

    let x = -123;

    println!("{:?}", reverse(x));
    println!("{}", std::i32::MAX > 1056389759)
}
